main.go

```diff
@@ -1,11 +1,50 @@
 package main
 
 import (
-	"./os/user"
 	"fmt"
+	"gogetent/os/user"
+	"sync"
 )
 
-func main(){
-	u, _ := user.Lookup("mjonaitis")
-	fmt.Printf("%+v", u)
+var wg = sync.WaitGroup{}
+
+func main() {
+	num := 3
+
+	for i := 0; i < num; i++ {
+		go iterateUsers(i)
+		wg.Add(1)
+	}
+	wg.Wait()
+
+	fmt.Println("Starting to iterate over groups")
+	for i := 0; i < num; i++ {
+		go iterateGroups(i)
+		wg.Add(1)
+	}
+	wg.Wait()
+}
+
+var mu = sync.Mutex{}
+
+func iterateUsers(listNum int) {
+	mu.Lock()
+	l := make([]string, 0, 10)
+	_ = user.IterateUsers(func(u *user.User) error {
+		l = append(l, u.Username)
+		return nil
+	})
+	fmt.Printf("Goroutine %d user names: %+v \n\n", listNum, l)
+	wg.Done()
```

Comment:
Any reason not to defer `wg.Done` and `mu.Unlock`?

Reply:
Please disregard main.go - at the moment, it is only used for testing how iteration functionality from os/user package works.